src/uct/sm/cma/cma_iface.c

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) UT-Battelle, LLC. 2014-2015. ALL RIGHTS RESERVED.
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
+ * $COPYRIGHT$
+ * $HEADER$
+ */
+
+#include "cma_pd.h"
+#include "cma_iface.h"
+#include "cma_ep.h"
+
+#include <uct/api/addr.h>
+#include <uct/tl/context.h>
+
+UCT_CMA_COMPONENT_DEFINE(uct_cma_pd, "cma", NULL)
+UCT_PD_REGISTER_TL(&uct_cma_pd, &uct_cma_tl);
+
+static ucs_config_field_t uct_cma_iface_config_table[] = {
+    {"", "ALLOC=huge,mmap,heap", NULL,
+    ucs_offsetof(uct_cma_iface_config_t, super),
+    UCS_CONFIG_TYPE_TABLE(uct_iface_config_table)},
+    {NULL}
+};
+
+static ucs_status_t uct_cma_iface_get_address(uct_iface_t *tl_iface,
+                                             struct sockaddr *addr)
+{
+    uct_sockaddr_process_t *iface_addr = (void*)addr;
+    iface_addr->sp_family = UCT_AF_PROCESS;
+    iface_addr->node_guid = ucs_machine_guid();
+    return UCS_OK;
+}
+
+static int uct_cma_iface_is_reachable(uct_iface_t *tl_iface,
+                                     const struct sockaddr *addr)
+{
+    return (addr->sa_family == UCT_AF_PROCESS) &&
+           (((uct_sockaddr_process_t*)addr)->node_guid == ucs_machine_guid());
+}
+
+static ucs_status_t uct_cma_flush()
+{
+    return UCS_OK;
+}
+
+static ucs_status_t uct_cma_iface_query(uct_iface_h tl_iface,
+                                       uct_iface_attr_t *iface_attr)
+{
+    memset(iface_attr, 0, sizeof(uct_iface_attr_t));
+
+    /* default values for all shared memory transports */
+    iface_attr->cap.put.max_zcopy      = 64*1024*1024;
```

Comment:
It was crushing on my laptop vmware :) I will add comment that this temporary threshold and we will update it later. I have to figure out what is the limit for cma.